app/main.py

```diff
@@ -1,3 +1,57 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[int, float, Distance]) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        return Distance(km=self.km + other)
+
+    def __iadd__(self, other: Union[int, float, Distance]) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        self.km += other
+        return self
+
+    def __mul__(self, other: Union[int, float, Distance]) -> Distance:
+        return Distance(km=self.km * other)
+
+    def __truediv__(self, other: Union[int, float]) -> Distance:
+        self.km = round(self.km / other, 2)
+        return self
```

Comment:
The `__truediv__` method should handle division by zero to prevent runtime errors. Consider adding a check to raise an exception or handle this case appropriately.